include/patchestry/Passes/PatchSpec.hpp

```diff
@@ -381,6 +382,7 @@ namespace llvm::yaml {
             io.mapOptional("index", arg.index);
             io.mapOptional("symbol", arg.symbol);
             io.mapOptional("value", arg.value);
+            io.mapOptional("is_reference", arg.is_reference);
```

Comment:
General question: I may be misunderstanding the YAML IO field mapping API, but why are we using `mapOptional` for so many fields? It makes sense to me in this case, but doesn't it make configuration file parsing more permissive than necessary generally speaking?

Reply:
This is good question. We are more permissive with YAML fields. We can move to more restrictive and mark some of these fields as required.

Reply:
@kaoudis, I marked some of the yaml fields required to make sure it fails if it is missing.